### PR TITLE
Add a config variable to override default container registry instead of dockerhub. (#10166)

### PR DESCRIPTION
**Pull request by liushilongbuaa**
#### Why I did it
* Add variable to reset default docker registry
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

**Comment by xumia** [comment]
@liushilongbuaa , please update the description.

**Review by xumia — APPROVED** [review]

**Comment by liushilongbuaa** [comment]
> @liushilongbuaa , please update the description.

Updated, I will use rebase merge.

File: Makefile.work [hunk 1]
```diff
@@ -115,6 +115,10 @@ rules/config.user:
 include rules/config
 -include rules/config.user
 
+ifneq ($(DEFAULT_CONTAINER_REGISTRY),)
+override DEFAULT_CONTAINER_REGISTRY := $(DEFAULT_CONTAINER_REGISTRY)/
```
Comment (qiluo-msft) [inline Makefile.work:119]: "container registry" is a weird name. Do you mean [docker registry](https://docs.docker.com/registry/) ?

Comment (liushilongbuaa) [reply Makefile.work:119]: yes, it seems docker registry is better.
container registry comes from azure container registry.
When I enable ACR in pipeline, it is a chance to change.

**Pull request merged**